### ISSUE TITLE
Questions with checkboxes - Roles provided for the checkboxes are not valid

### ISSUE THREAD

**Issue by ParthaA11y** [surveyjs/survey-library#8460]
**Original Issue:** [Accessibility issue - SurveyJS Player](https://surveyjs.answerdesk.io/internal/ticket/details/T11244).

**Title**: When Aria attributes are not used correctly assistive technology may not correctly function as expected. 

URL: https://surveyjs.io/form-library/examples/nps-question/reactjs

**Issue**: 

Role=listbox is used in the <fieldset> and role=option is used for the input field with type=checkbox. As a result Assistive technology such as screen reader does not recognize the correct role and the state of the checkbox and does not read them as checkbox. Also, JAWS is providing instruction to use arrow key to navigate but using arrow key is not working in this case.
A listbox and option are used in the dropdown or combo boxes. 
[Listbox example](https://www.w3.org/WAI/ARIA/apg/patterns/listbox/examples/listbox-grouped/)

**Expectation**: On focus screen reader should read the accessible name, valid role and current state of the element. For the checkbox screen reader should read the role as checkbox and current state as checked or not checked, so that a user can understand the functionality and keyboard interaction of the checkbox. 

**Code snippet**:

**How to fix**: Please remove the role=listbox from the fieldset and role=option from the inputs.

**Tools used**: Screen reader JAWS, NVDA.
**Success criteria**: WCAG 4.1.2: Name role and value 
https://www.w3.org/WAI/WCAG21/Techniques/general/G108
https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA4

**Screenshot:** 
![image](https://github.com/surveyjs/survey-library/assets/126093390/088cb5d9-3a33-4b02-9398-8040a38a6955)

**Screen recordings:** 

1. JAWS without role=listbox and option: [https://www.loom.com/share/9d1fa1574de044e19d26cc31cfd2e275?sid=e1f70162-5fdf-410d-b772-0b9dd7050cb3](https://nam02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fwww.loom.com%2Fshare%2F9d1fa1574de044e19d26cc31cfd2e275%3Fsid%3De1f70162-5fdf-410d-b772-0b9dd7050cb3&data=05%7C02%7Cpartha.sarathi.das%40walgreens.com%7C7f8d22d0c5ff483500bd08dc949aa31a%7C92cb778e8ba74f34a0114ba6e7366996%7C0%7C0%7C638548639079214560%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C0%7C%7C%7C&sdata=2LrDaupo9Dts1E85aQXa03jm8Zl4wyyt41g7OefLQyw%3D&reserved=0) 

2. NVDA Without role=listbox and option: [https://www.loom.com/share/f1548edd0574487eb712574e91fb6f86?sid=c8354f10-9227-4713-b1a5-6e9ae6258362](https://nam02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fwww.loom.com%2Fshare%2Ff1548edd0574487eb712574e91fb6f86%3Fsid%3Dc8354f10-9227-4713-b1a5-6e9ae6258362&data=05%7C02%7Cpartha.sarathi.das%40walgreens.com%7C7f8d22d0c5ff483500bd08dc949aa31a%7C92cb778e8ba74f34a0114ba6e7366996%7C0%7C0%7C638548639079224769%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C0%7C%7C%7C&sdata=9WI5YmrUcKKxeLYRd%2F%2FMiDuazjHy533U%2BIQxelKccAc%3D&reserved=0)

3. NVDA with listbox and option: [https://www.loom.com/share/bc99a8e085bf4a10ab7949b693bd6455?sid=af1910f8-203b-4ffc-b1f9-51acf82df6c5](https://nam02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fwww.loom.com%2Fshare%2Fbc99a8e085bf4a10ab7949b693bd6455%3Fsid%3Daf1910f8-203b-4ffc-b1f9-51acf82df6c5&data=05%7C02%7Cpartha.sarathi.das%40walgreens.com%7C7f8d22d0c5ff483500bd08dc949aa31a%7C92cb778e8ba74f34a0114ba6e7366996%7C0%7C0%7C638548639079231796%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C0%7C%7C%7C&sdata=Egi14X0RWznA%2FzfE1RWVPWK85Q4Z23GZgiTqE9qmAmA%3D&reserved=0)

4. JAWS with listbox and option: [https://www.loom.com/share/43fff5b629134010a45a21189c58e8c8?sid=8c555f4e-53bc-4f48-92b9-c639519517f5](https://nam02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fwww.loom.com%2Fshare%2F43fff5b629134010a45a21189c58e8c8%3Fsid%3D8c555f4e-53bc-4f48-92b9-c639519517f5&data=05%7C02%7Cpartha.sarathi.das%40walgreens.com%7C7f8d22d0c5ff483500bd08dc949aa31a%7C92cb778e8ba74f34a0114ba6e7366996%7C0%7C0%7C638548639079238741%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C0%7C%7C%7C&sdata=grr0PjczFQB2qTU4mQQHB%2FZZaEOjfQfMKt5Kucod208%3D&reserved=0)

**Comment by dmitry-kurmanov** [comment]
related to https://github.com/surveyjs/survey-library/pull/5269/files
